Common/Source/Calc/WaypointArrivalAltitude.cpp

```diff
@@ -18,6 +18,11 @@ void simpleETE(NMEA_INFO *Basic, DERIVED_INFO *Calculated, int i) {
    if (!ValidWayPoint(i)) return;
 
    WayPointCalc[i].NextETE= WayPointCalc[i].Distance / Basic->Speed;
+   if (Calculated->AverageGS>0) {
+       WayPointCalc[i].NextAvrETE = WayPointCalc[i].Distance / Calculated->AverageGS;
+   } else {
+       WayPointCalc[i].NextAvrETE= WayPointCalc[i].NextETE;
```

Comment:
could be better to set negative value, like that, you draw "---" in the info box if no Avg ground speed is available...

Reply:
"---" until no data available is better than a value with other meaning....

Reply:
Right, changed.